src/main/java/uk/gov/digital/ho/hocs/workflow/api/bpmn/BusinessEventService.java

```diff
@@ -0,0 +1,98 @@
+
+@Service
+@Slf4j
+public class BusinessEventService {
+    private final AuditClient auditClient;
+    private final CaseworkClient caseworkClient;
+    private final BpmnService bpmnService;
+    private final FormService formService;
+
+    public BusinessEventService(AuditClient auditClient,
+                                CaseworkClient caseworkClient,
+                                BpmnService bpmnService,
+                                FormService formService) {
+        this.auditClient = auditClient;
+        this.caseworkClient = caseworkClient;
+        this.bpmnService = bpmnService;
+        this.formService = formService;
+    }
+
+    public void createDataFieldUpdatedEvent(String caseUUIDString, String stageUUIDString, String screenName, String fieldName, String newValue) {
+        UUID caseUUID = UUID.fromString(caseUUIDString);
+        UUID stageUUID= UUID.fromString(stageUUIDString);
+
+        String previousValueKey = String.format(PREVIOUS_VALUE_KEY, fieldName);
+        String name = String.format(DATA_FIELD_UPDATED_EVENT_NAME, fieldName);
+
+        GetCaseworkCaseDataResponse caseData = caseworkClient.getCase(caseUUID);
+        String previousValue = caseData.getData().get(previousValueKey);
+
+        if (Objects.equals(previousValue, newValue)) {
+            return;
+        }
+
+        HocsSchema hocsSchema = formService.getFormSchema(screenName);
+
+        List<HocsFormField> fields = hocsSchema.getFields();
+        Optional<HocsFormField> hocsFormField = fields.stream().filter((field) -> Objects.equals(field.getProps().get("name"), fieldName)).findFirst();
+
+        String fieldNameLabel = getFormProperty("label", hocsFormField, fieldName, String.class);
+
+        @SuppressWarnings("unchecked") List<Map<String, String>> choices = getFormProperty("choices", hocsFormField, List.of(), List.class);
+
+        String newValueLabel = getLabel(newValue, choices);
+        String previousValueLabel = getLabel(previousValue, choices);
+
+
+        DataFieldUpdatedPayload dataFieldUpdatedPayload = new DataFieldUpdatedPayload(
+            name,
+            fieldName,
+            newValue,
+            previousValue,
+            fieldNameLabel,
+            newValueLabel,
+            previousValueLabel
+        );
+
+        auditClient.createBusinessEvent(caseUUID, stageUUID, dataFieldUpdatedPayload);
+        bpmnService.updateCaseValue(caseUUIDString, stageUUIDString, previousValueKey, newValue);
+    }
+
+    private String getLabel(String value, List<Map<String, String>> choices) {
+        return choices.stream()
+            .filter(choice -> Objects.equals(choice.get("value"), value))
+            .findFirst()
+            .map(choice -> choice.get("label"))
+            .orElse(value);
+    }
+
+    private <T> T getFormProperty(String name, Optional<HocsFormField> hocsFormField, T defaultValue, Class<T> classTag) {
+        return hocsFormField
```

Comment:
This needs to take conditional values into account. You need to check for conditional field props and use that if there is one, otherwise default to the non conditional field.